cmd/xk6/main.go

```diff
@@ -30,123 +29,66 @@ import (
 )
 
 var (
-	k6Version    = os.Getenv("K6_VERSION")
-	k6Repo       = os.Getenv("XK6_K6_REPO")
-	raceDetector = os.Getenv("XK6_RACE_DETECTOR") == "1"
-	skipCleanup  = os.Getenv("XK6_SKIP_CLEANUP") == "1"
+	errExpectedValue  = fmt.Errorf("expected value")
+	errInvalidValue   = fmt.Errorf("invalid value")
+	errMissingFlag    = fmt.Errorf("missing flag")
+	errMissingReplace = fmt.Errorf("missing replace")
 )
 
+type BuildOps struct {
+	K6Version      string
+	Extensions     []string
+	Replacements   []string
+	OutFile        string
+	OutputOverride bool
+}
+
 func main() {
+	log := slog.New(
+		slog.NewTextHandler(
+			os.Stderr,
+			&slog.HandlerOptions{
+				Level: slog.LevelDebug,
+			},
+		),
+	)
+
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	go trapSignals(ctx, cancel)
+	go trapSignals(ctx, log, cancel)
 
 	if len(os.Args) > 1 && os.Args[1] == "build" {
-		if err := runBuild(ctx, os.Args[2:]); err != nil {
-			log.Fatalf("[ERROR] %v", err)
+		if err := runBuild(ctx, log, os.Args[2:]); err != nil {
+			log.Error(fmt.Sprintf("build error %v", err))
 		}
 		return
 	}
 
-	if err := runDev(ctx, os.Args[1:]); err != nil {
-		log.Fatalf("[ERROR] %v", err)
+	if err := runDev(ctx, log, os.Args[1:]); err != nil {
+		log.Error(fmt.Sprintf("run error %v", err))
```

Comment:
Originally, there was log.Fatal here, which resulted in an exit status of 1. Currently, an exit status of 0 is generated in case of an error, which is a problem from an automation perspective.